### PR TITLE
SW-1963 Reflect changes in grblHAL core module

### PR DESCRIPTION
**Pull request by amrithvenkat**
* Include corexy.c in the cmakelists.
* Enable the define for corexy in config.h.

### CODE REVIEW

File: config.h [hunk 1]
```diff
@@ -529,8 +529,8 @@ __NOTE:__ these definitions are only referenced in this file. Do __NOT__ change!
 //#define DEFAULT_X_ACCELERATION (10.0*60*60) // 10*60*60 mm/min^2 = 10 mm/sec^2
 //#define DEFAULT_Y_ACCELERATION (10.0*60*60) // 10*60*60 mm/min^2 = 10 mm/sec^2
 //#define DEFAULT_Z_ACCELERATION (10.0*60*60) // 10*60*60 mm/min^2 = 10 mm/sec^2
-//#define DEFAULT_X_MAX_TRAVEL 200.0f // mm NOTE: Must be a positive value.
-//#define DEFAULT_Y_MAX_TRAVEL 200.0f // mm NOTE: Must be a positive value.
+#define DEFAULT_X_MAX_TRAVEL 507.0f // mm NOTE: Must be a positive value.
```
Comment (ahmed-mrbeam) [inline config.h:532]: from where we got these numbers, Please mention in code comment

File: config.h [hunk 2]
```diff
@@ -529,8 +529,8 @@ __NOTE:__ these definitions are only referenced in this file. Do __NOT__ change!
...
+#define DEFAULT_Y_MAX_TRAVEL 390.0f // mm NOTE: Must be a positive value.
 //#define DEFAULT_Z_MAX_TRAVEL 200.0f // mm NOTE: Must be a positive value.
```
Comment (ahmed-mrbeam) [inline config.h:534]: Don't we need that for MR Beam 3 ?

File: config.h [hunk 3]
```diff
@@ -495,7 +495,7 @@ __NOTE:__ these definitions are only referenced in this file. Do __NOT__ change!
 //#define DEFAULT_NO_REPORT_WORK_COORD_OFFSET
 //#define DEFAULT_NO_REPORT_OVERRIDES
 //#define DEFAULT_REPORT_PARSER_STATE
```
[the remark this answers is not in the document]
Comment (amrithvenkat) [reply config.h:497]: CoreXY is the module that is used for the Gantry. https://corexy.com/

File: grbllib.c [hunk 4]
```diff
@@ -266,7 +267,7 @@ int grbl_enter (void)
 
         // Print welcome message. Indicates an initialization has occurred at power-up or with a reset.
         report_init_message();
-
+        watchdog_update();
```
Comment (ahmed-mrbeam) [inline grbllib.c:270]: Please mention in code comment that the watchdog config used here relies on the Bootloader config for the watchdog

**Review by ahmed-mrbeam — COMMENTED** [review]
Added some comments, kindly check

File: config.h (hunk 2 above)
Comment (amrithvenkat) [reply config.h:534]: We need the exact value when the Engineering team finalizes the working area limits for the Mr Beam 3. currently it is set to the limits that are for the existing Mr Beam devices.

File: grbllib.c (hunk 4 above)
Comment (amrithvenkat) [reply grbllib.c:270]: okay

File: config.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (amrithvenkat) [reply config.h:497]: Enabling it is the same as sending a $G command to the terminal during the lasering job. 

Output on old device with grbl 0.9 looks like
15:27:46,864 _COMM_: Send: $G*107
15:27:46,874 _COMM_: Recv: [G1 G54 G17 G21 G90 G94 M0 M5 M9 T0 F5000. S0.] 
 
Same in the grblHAL version. Enabling it via the #define prints out the parser state only when there is a change

File: grbllib.c (hunk 4 above)
Comment (amrithvenkat) [reply grbllib.c]: done

**Review by ahmed-mrbeam — APPROVED** [review]

**Pull request merged**